### ISSUE TITLE
Tracking of subscription requests

### ISSUE THREAD

**Issue by jonkri** [pontarius/pontarius-service#17]
We need to know which subscription requests we have yet to answer in some way. Here are some initial thoughts about this.

I guess that if a peer is added to a contact, the subscription request is granted automatically. Hence, when a subscription request arrives, we probably deny it and ignore the peer, or add the peer to a contact (and have Pontarius Service accept the request on our behalf). (I don't think that it's meaningful to accept the request without adding the peer to a contact, but I'm not sure.)

If we reason like above, it's not about subscription requests anymore, but about discovering peers and letting the link/ignore actions determine what to do with them. Either way, the delta pattern could be the way to go, so that other Pontarius applications get notified about what has happened. (For information about deltas, see https://github.com/pontarius/pontarius-service/issues/4#issuecomment-149625200.)

**Comment by jonkri** [comment]
@Philonous: I'm posting this one without any particular thought about the HuBoard order as well.

**Comment by jonkri** [comment]
Technically, we could accept the request and at the same time either link or ignore the user. We could also deny the request and either remove the peer, leave it as unlinked, or simply ignore it. I don't think the UI will need to support all these possible cases though. If the user is only using Pontarius Service for XMPP, the binary choice mentioned in the issue makes sense. If not, perhaps it's reasonable to include the "accept and ignore" option in addition to the two original choices.

**Comment by jonkri** [comment]
In a way, the choices should probably be the same as having an "unassigned/unlinked" peer from which we have <em>not</em> received a subscription request.

**Comment by jonkri** [comment]
Maybe we can do it like this.
- A presence subscription request from an unlinked peer doesn't mean anything. It's simply not visible to the user. The assumption is that the user will decide what to do with any peer that is known but not linked or ignored.
- A presence subscription request from an ignored peer results in a dialog which allows the user to specify whether to accept and reject the request.

In other words, when the user decides to ignore a peer, they get a question about what to do with the request.

**Comment by jonkri** [comment]
Talk about rambling. :)

**Comment by jonkri** [comment]
Linking the peer and accepting the subscription request become the same thing.
